### ISSUE TITLE
Which one can be used for both classification and regression?

### ISSUE THREAD

**Issue by izxi** [izxi/Learning#19]
In the machining learning methods listed below, which one can be used for both classification and regression?

Select one:

- A. Linear Regression
- B. Support Vector Machines
- C. Random Forests
- D. Decision Trees

**Comment by izxi** [comment]
In machine learning, **support vector machines** (SVMs, also support vector networks[1]) are supervised learning models with associated learning algorithms that analyze data used for classification and regression analysis. Given a set of training examples, each marked as belonging to one or the other of two categories, an SVM training algorithm builds a model that assigns new examples to one category or the other, making it a non-probabilistic binary linear classifier (although methods such as Platt scaling exist to use SVM in a probabilistic classification setting). An SVM model is a representation of the examples as points in space, mapped so that the examples of the separate categories are divided by a clear gap that is as wide as possible. New examples are then mapped into that same space and predicted to belong to a category based on which side of the gap they fall.

In addition to performing linear classification, SVMs can efficiently perform a non-linear classification using what is called the kernel trick, implicitly mapping their inputs into high-dimensional feature spaces.

When data are not labeled, supervised learning is not possible, and an unsupervised learning approach is required, which attempts to find natural clustering of the data to groups, and then map new data to these formed groups. The support vector clustering[2] algorithm created by Hava Siegelmann and Vladimir Vapnik, applies the statistics of support vectors, developed in the support vector machines algorithm, to categorize unlabeled data, and is one of the most widely used clustering algorithms in industrial applications.

**Comment by izxi** [comment]
[Decision Trees](http://scikit-learn.org/stable/modules/tree.html) (DTs) are a non-parametric supervised learning method used for classification and regression. The goal is to create a model that predicts the value of a target variable by learning simple decision rules inferred from the data features.

**Comment by izxi** [comment]
[Support vector machines](http://scikit-learn.org/stable/modules/svm.html) (SVMs) are a set of supervised learning methods used for classification, regression and outliers detection.

**Comment by izxi** [comment]
A **random forest** is a meta estimator that fits a number of decision tree classifiers on various sub-samples of the dataset and use averaging to improve the predictive accuracy and control over-fitting. The sub-sample size is always the same as the original input sample size but the samples are drawn with replacement if bootstrap=True (default).

**Comment by rysbekova** [comment]
D
https://machinelearningmastery.com/classification-versus-regression-in-machine-learning/

**Comment by johnnieng** [comment]
C. Random Forests *

**Comment by om-ganesh** [comment]
@johnnieng You have asterik(*) in some of your solution ... What does that mean? Does that mean you are confident of the answer or have still some doubts?

**Comment by johnnieng** [comment]
> 
> 
> @johnnieng You have asterik(*) in some of your solution ... What does that mean? Does that mean you are confident of the answer or have still some doubts?

Confident

Random forests or random decision forests are an ensemble learning method for classification, regression and other tasks that operate by constructing a multitude of decision trees at training time and outputting the class that is the mode of the classes (classification) or mean prediction (regression) of the individual trees.

**Comment by LSD20-crypto** [comment]
It's decision tree.
100% sure.

**Comment by LSD20-crypto** [comment]
Support vector machine SVM

**Comment by mebriki** [comment]
**_D. Decision trees_** (most probable answer)
[Classification vs Regression:](https://machinelearningmastery.com/classification-versus-regression-in-machine-learning/)
Some algorithms **_can be used for both classification and regression_** with small modifications, such as **_decision trees_** and artificial neural networks. 
Some algorithms **_cannot, or cannot easily be used_** for both problem types, such as **_linear regression_** for regression predictive modeling and logistic regression for classification predictive modeling.
**_SVMs_** although can be used for both, they are mostly used for **_classification_**
